socketio/src/client/reconnect.rs

```diff
@@ -0,0 +1,216 @@
+#![allow(unused)]
+use std::{
+    ops::Deref,
+    sync::{Arc, Mutex, RwLock},
+    time::Duration,
+};
+
+pub use super::super::{event::Event, payload::Payload};
+use super::{Client, ClientBuilder};
+use crate::{error::Result, packet::Packet, Error};
+use backoff::ExponentialBackoff;
+use backoff::{backoff::Backoff, ExponentialBackoffBuilder};
+
+#[derive(Clone)]
+pub struct ReconnectClient {
+    builder: ClientBuilder,
+    client: Arc<RwLock<super::Client>>,
+    backoff: ExponentialBackoff,
+    max_reconnect_attempts: u8,
+}
+
+impl ReconnectClient {
+    pub(crate) fn new(builder: ClientBuilder) -> Result<Self> {
+        let builder_clone = builder.clone();
+        let client = builder_clone.connect_manual()?;
+        let backoff = ExponentialBackoffBuilder::new()
+            .with_initial_interval(Duration::from_millis(builder.reconnect_delay))
+            .with_max_interval(Duration::from_millis(builder.reconnect_delay_max))
+            .build();
+        let max_reconnect_attempts = builder.reconnect_attempts;
```

Comment:
Why do we need this variable here? Can't we just ise the builder in the Self construction?

Reply:
You are right!